pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authentication/OneStageAuthenticationState.java

```diff
@@ -35,20 +38,33 @@
 public class OneStageAuthenticationState implements AuthenticationState {
 
     private final AuthenticationDataSource authenticationDataSource;
-    private final String authRole;
+    private List<String> authRoles;
 
     public OneStageAuthenticationState(AuthData authData,
                                        SocketAddress remoteAddress,
                                        SSLSession sslSession,
                                        AuthenticationProvider provider) throws AuthenticationException {
         this.authenticationDataSource = new AuthenticationDataCommand(
             new String(authData.getBytes(), UTF_8), remoteAddress, sslSession);
-        this.authRole = provider.authenticate(authenticationDataSource);
+        try {
+            this.authRoles = provider.authenticate(authenticationDataSource, true);
```

Comment:
We should check multi is implemented and then call the appropriated method. This change will cause users who have custom implementations to always throw an exception

Reply:
I think this is OK. When multi roles authenticate is not supported, the original authenticate method will be called.

Reply:
This will always throw an exception with older providers and I think it's better to have a flag instead of checking a string value in an exception.